nupic/research/frameworks/vernon/experiments/self_supervised_experiment.py

```diff
@@ -0,0 +1,452 @@
+        """
+        Creates a dataloader for the supervised training data.
+
+        This method is a class method so that it can be used directly by
+        analysis
+        tools, while also being easily overrideable.
+        """
+
+        sampler = cls.create_supervised_sampler(config, dataset)
+        return DataLoader(
+            dataset=dataset,
+            batch_size=config.get("batch_size_supervised", 1),
+            shuffle=sampler is None,
+            num_workers=config.get("workers", 0),
+            sampler=sampler,
+            pin_memory=torch.cuda.is_available(),
+            drop_last=config.get("supervised_loader_drop_last", True),
+        )
+
+    @classmethod
+    def create_validation_dataloader(cls, config, dataset):
+        """
+        Creates a dataloader for the validation data.
+
+        This method is a class method so that it can be used directly by analysis
+        tools, while also being easily overrideable.
+        """
+        sampler = cls.create_validation_sampler(config, dataset)
+        return DataLoader(
+            dataset=dataset,
+            batch_size=config.get("val_batch_size", config.get("batch_size", 1)),
+            shuffle=False,
+            num_workers=config.get("workers", 0),
+            sampler=sampler,
+            pin_memory=torch.cuda.is_available(),
+        )
+
+    @classmethod
+    def create_unsupervised_sampler(cls, config, dataset):
+        num_samples = config.get("num_unsupervised_samples", -1)
+        if num_samples > 0:
+            return RandomSampler(dataset, num_samples=num_samples)
+        return None
+
+    @classmethod
+    def create_supervised_sampler(cls, config, dataset):
+        num_samples = config.get("num_supervised_samples", -1)
+        if num_samples > 0:
+            return RandomSampler(dataset, num_samples=num_samples)
+        return None
+
+    @classmethod
+    def create_validation_sampler(cls, config, dataset):
+        num_samples = config.get("num_validation_samples", -1)
+        if num_samples > 0:
+            return RandomSampler(dataset, num_samples=num_samples)
+        return None
+
+    @classmethod
+    def load_dataset(cls, config, train=True, dataset_type="unsupervised"):
+        """
+        Loads one of three types of datasets: unsupervised, supervised, and validation
+
+        The dataset_class argument in the config file should either be a single
+        dataset class, or a dict with the following structure:
+        {
+         "unsupervised": unsupervised_dataset_class
+         "supervised": supervised_dataset_class
+         "validation": validation_dataset_class
+        }
+
+        Similarly, the dataset_args argument in the config should be a dict of dicts
+        with keys corresponding to the different dataset types as described above.
+        """
+        dataset_class = config.get("dataset_class", None)
+        dataset_args = dict(config.get("dataset_args", {}))
```

Comment:
Why do you typecast this even though it's meant to be a dict?

Reply:
This was actually following suit from the SupervisedExperiment method. I was also a bit confused but went with it, though it does seem unnecessary, so I removed it. Should also probably be changed in SupervisedExperiment.